src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_producer_power_formula.py

```diff
@@ -43,35 +50,111 @@ def generate(  # noqa: DOC502
         )
 
         component_graph = connection_manager.get().component_graph
-        # if in the future we support additional producers, we need to add them to the lambda
-        producer_components = component_graph.dfs(
-            self._get_grid_component(),
-            set(),
-            lambda component: component_graph.is_pv_chain(component)
-            or component_graph.is_chp_chain(component),
-        )
-
-        if not producer_components:
-            _logger.warning(
-                "Unable to find any producer components in the component graph. "
-                "Subscribing to the resampling actor with a non-existing "
-                "component id, so that `0` values are sent from the formula."
+        if self._config.component_ids is None:
```

Comment:
why do we need to check this for producer power formula?

Reply:
simplified in last commit